File: Keepr.client/src/models/VaultKeep.js
import { Keep } from "./Keep.js"
import { Vault } from "./Vault.js"

export class VaultKeep {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.vaultId = new Vault(data.vaultId)
    this.keepId = new Keep(data.keepId)
  }
}

File: Keepr/Models/VaultKeep.cs
using Keepr.Interfaces;

namespace Keepr.Models;

public class VaultKeep : ICreated, IRepoItem<int>
{
  public int Id { get; set; }

  public string CreatorId { get; set; }
  public int VaultId { get; set; }
  public int KeepId { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public Profile Creator { get; set; }
}

